Automattic/jetpack
# JITM: Sideload Jetpack Backup

#### Changes proposed in this Pull Request:

Adds the logic necessary to enable installing and activating the Jetpack Backup plugin via JITM. It mimics the behavior and code pattern of the CreativeMail JITM sideloading. 

This is in preparation of enabling the JITM in wpcom later when we're ready, so that we don't need to wait for another release. The wpcom diff linked below is not up for review, but is useful for testing the PR. 

#### Jetpack product discussion


p1HpG7-dzq-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






Set up: 
- Connect to wpcom sandbox
- Apply D69991-code to sandbox
- Make sure Jetpack Backup is not installed on the Jetpack site
- You'll see the JITM on the main wp-admin dashboard as well as the Jetpack dashboard. 

Successful install: 
- Clicking it should install/activate and navigate you to the Jetpack Backup plugin UI. 
- ~Check to see if the /s/jetpack-jitm-plugin-install/ MC stat was bumped properly~ doing this in separate PR

Failed install: 
- Force a failure in try_install(). I tested by manually modifying the code. 
- Click the JITM to install. You should get redirected to the plugin search page in wp-admin with jetpack-backup results pre-populated. 
- You should see a wp-admin notice telling you about the failure. 
- ~Check to see if the /s/jetpack-jitm-plugin-install/ MC stat was bumped properly~ doing this in separate PR
